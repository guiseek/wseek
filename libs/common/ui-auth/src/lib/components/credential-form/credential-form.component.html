<mat-card>
  <mat-card-title>Login</mat-card-title>
  <mat-card-content>
    <form [formGroup]="form" (ngSubmit)="submit()">
      <p>
        <mat-form-field>
          <input
            type="text"
            matInput
            placeholder="Endereço de email"
            formControlName="email"
          />
          <mat-error *ngIf="email.invalid">
            <span *ngIf="email.hasError('required')">Email obrigatório</span>
            <span *ngIf="email.hasError('email')">Email inválido</span>
          </mat-error>
        </mat-form-field>
      </p>

      <p>
        <mat-form-field>
          <input
            type="password"
            matInput
            placeholder="Senha"
            formControlName="password"
          />
          <mat-error *ngIf="password.invalid">
            <span *ngIf="password.hasError('required')">Senha obrigatória</span>
            <span *ngIf="password.hasError('minlength')">Senha muito curta</span>
          </mat-error>
        </mat-form-field>
      </p>

      <mat-error *ngIf="errorMessage">
        {{ errorMessage }}
      </mat-error>

      <p class="loginButtons">
        <button type="submit" mat-button>Login</button>
      </p>
    </form>
  </mat-card-content>
</mat-card>
