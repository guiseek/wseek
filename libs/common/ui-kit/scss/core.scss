// Este arquivo deve ser importado apenas uma vez!
// Use seek.scss para acessar Seek/Material mixins

// Perfect scrollbar
@import '~perfect-scrollbar/css/perfect-scrollbar';

// Seek
@import 'seek';

// Material styles
@include mat-core();

// Mixins
@import 'mixins/colors';
@import 'mixins/material';
@import 'mixins/typography';
@import 'mixins/page-layouts';
@import 'mixins/cards';
@import 'mixins/forms';

// Partials
@import 'partials/normalize';
@import 'partials/scrollbars';
@import 'partials/helpers';
@import 'partials/general';
@import 'partials/icons';
@import 'partials/angular-material-fix';
@import 'partials/docs';
@import 'partials/navigation';
@import 'partials/print';

// Plugins
@import 'plugins/plugins';
