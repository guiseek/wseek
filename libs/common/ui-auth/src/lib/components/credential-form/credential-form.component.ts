import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
import { FormControl, FormGroup, Validators } from '@angular/forms';

const Validator = {
  email: [Validators.required, Validators.email],
  password: [Validators.required, Validators.minLength(6)],
};

@Component({
  selector: 'seek-credential-form',
  templateUrl: './credential-form.component.html',
  styleUrls: ['./credential-form.component.scss'],
})
export class CredentialFormComponent {
  @Input() set pending(isPending: boolean) {
    if (isPending) {
      this.form.disable();
    } else {
      this.form.enable();
    }
  }

  @Input() errorMessage!: string | null;

  @Output() submitted = new EventEmitter<Credential>();

  form: FormGroup = new FormGroup({
    email: new FormControl('', Validator.email),
    password: new FormControl('', Validator.password),
  });

  get email() {
    return this.form.get('email');
  }

  get password() {
    return this.form.get('password');
  }

  submit() {
    this.form.markAllAsTouched();
    if (this.form.valid) {
      this.submitted.emit(this.form.value);
    }
  }
}
