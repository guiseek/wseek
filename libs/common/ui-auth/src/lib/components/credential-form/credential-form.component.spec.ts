import { ReactiveFormsModule } from '@angular/forms';
import { ComponentFixture, TestBed } from '@angular/core/testing';

import { CredentialFormComponent } from './credential-form.component';
import { NO_ERRORS_SCHEMA } from '@angular/core';
import * as axe from 'axe-core';


describe('CredentialFormComponent', () => {
  let component: CredentialFormComponent;
  let fixture: ComponentFixture<CredentialFormComponent>;

  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [ReactiveFormsModule],
      declarations: [CredentialFormComponent],
      schemas: [NO_ERRORS_SCHEMA],
    });

    fixture = TestBed.createComponent(CredentialFormComponent);
    component = fixture.componentInstance;
  });

  it('should compile', () => {
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('should disable form if pending', () => {
    component.pending = true;

    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('should display error message if provided', () => {
    component.errorMessage = 'Credenciais invÃ¡lidas';

    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('should emit an event if the form is valid when submitted', () => {
    const credentials = {
      email: 'seek@ubuntu.gui',
      password: '123456',
    };
    component.form.setValue(credentials);

    spyOn(component.submitted, 'emit');
    component.submit();

    expect(component.submitted.emit).toHaveBeenCalledWith(credentials);
  });

  it('should a11y ok', () => {
    axe.configure({
      locale: {
        lang: 'pt-br',
        rules: {

          // accesskeys: {
          //   help: 'Der Wert des accesskey-Attributes muss einzigartig sein.'
          // }
          // ...
        },
        checks: {

          // 'aria-errormessage': {
          //   pass: {
          //     a: ''
          //   },
          //   // Note: doT (https://github.com/olado/dot) templates are supported here.
          //   fail:
          //     {

          //     }
          // }
          // ...
        }
      }
    })
    expect(component).toBeTruthy();
  });
});
