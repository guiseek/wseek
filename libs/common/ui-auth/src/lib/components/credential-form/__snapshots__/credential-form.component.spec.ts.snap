// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CredentialFormComponent should compile 1`] = `
<seek-credential-form
  form={[Function FormGroup]}
  submitted={[Function EventEmitter_]}
>
  <mat-card>
    <mat-card-title>
      Login
    </mat-card-title>
    <mat-card-content>
      <form
        class="ng-untouched ng-pristine ng-invalid"
        novalidate=""
      >
        <p>
          <mat-form-field>
            <input
              class="ng-untouched ng-pristine ng-invalid"
              formcontrolname="email"
              matinput=""
              placeholder="Endereço de email"
              type="text"
            />
            
            <mat-error>
              
              <span>
                Email obrigatório
              </span>
              
            </mat-error>
          </mat-form-field>
        </p>
        <p>
          <mat-form-field>
            <input
              class="ng-untouched ng-pristine ng-invalid"
              formcontrolname="password"
              matinput=""
              placeholder="Senha"
              type="password"
            />
            
            <mat-error>
              
              <span>
                Senha obrigatória
              </span>
              
            </mat-error>
          </mat-form-field>
        </p>
        
        <p
          class="loginButtons"
        >
          <button
            mat-button=""
            type="submit"
          >
            Login
          </button>
        </p>
      </form>
    </mat-card-content>
  </mat-card>
</seek-credential-form>
`;

exports[`CredentialFormComponent should disable form if pending 1`] = `
<seek-credential-form
  form={[Function FormGroup]}
  submitted={[Function EventEmitter_]}
>
  <mat-card>
    <mat-card-title>
      Login
    </mat-card-title>
    <mat-card-content>
      <form
        class="ng-untouched ng-pristine"
        novalidate=""
      >
        <p>
          <mat-form-field>
            <input
              class="ng-untouched ng-pristine"
              disabled=""
              formcontrolname="email"
              matinput=""
              placeholder="Endereço de email"
              type="text"
            />
            
          </mat-form-field>
        </p>
        <p>
          <mat-form-field>
            <input
              class="ng-untouched ng-pristine"
              disabled=""
              formcontrolname="password"
              matinput=""
              placeholder="Senha"
              type="password"
            />
            
          </mat-form-field>
        </p>
        
        <p
          class="loginButtons"
        >
          <button
            mat-button=""
            type="submit"
          >
            Login
          </button>
        </p>
      </form>
    </mat-card-content>
  </mat-card>
</seek-credential-form>
`;

exports[`CredentialFormComponent should display error message if provided 1`] = `
<seek-credential-form
  errorMessage={[Function String]}
  form={[Function FormGroup]}
  submitted={[Function EventEmitter_]}
>
  <mat-card>
    <mat-card-title>
      Login
    </mat-card-title>
    <mat-card-content>
      <form
        class="ng-untouched ng-pristine ng-invalid"
        novalidate=""
      >
        <p>
          <mat-form-field>
            <input
              class="ng-untouched ng-pristine ng-invalid"
              formcontrolname="email"
              matinput=""
              placeholder="Endereço de email"
              type="text"
            />
            
            <mat-error>
              
              <span>
                Email obrigatório
              </span>
              
            </mat-error>
          </mat-form-field>
        </p>
        <p>
          <mat-form-field>
            <input
              class="ng-untouched ng-pristine ng-invalid"
              formcontrolname="password"
              matinput=""
              placeholder="Senha"
              type="password"
            />
            
            <mat-error>
              
              <span>
                Senha obrigatória
              </span>
              
            </mat-error>
          </mat-form-field>
        </p>
        
        <mat-error>
           Credenciais inválidas 
        </mat-error>
        <p
          class="loginButtons"
        >
          <button
            mat-button=""
            type="submit"
          >
            Login
          </button>
        </p>
      </form>
    </mat-card-content>
  </mat-card>
</seek-credential-form>
`;
